src/common/models/dynamic-subset-formula/dynamic-subset-formula.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017-2020 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IncomingHttpHeaders } from "http";
+import { Instance } from "immutable-class";
+import { Expression } from "plywood";
+import { Unary } from "../../utils/functional/functional";
+
+export type DynamicSubsetFormulaFn = Unary<IncomingHttpHeaders, Expression>;
```

Comment:
What values could we pass there? Full request object? Only headers? Maybe more fields on request object like subdomains, path, ip etc?

Reply:
let's start from headers and make design extensible

Reply:
We need to pass plywood dependency, so function can use it's DSL and return real Expression